src/core/server/saved_objects/migrationsv2/integration_tests/migration_7_13_0_transform_failures.test.ts

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import Path from 'path';
+import Fs from 'fs';
+import Util from 'util';
+import * as kbnTestServer from '../../../../test_helpers/kbn_server';
+import { Root } from '../../../root';
+
+const logFilePath = Path.join(__dirname, '7_13_corrupt_transform_failures_test.log');
+
+const asyncUnlink = Util.promisify(Fs.unlink);
+async function removeLogFile() {
+  // ignore errors if it doesn't exist
+  await asyncUnlink(logFilePath).catch(() => void 0);
+}
+
+describe('migration v2', () => {
+  let esServer: kbnTestServer.TestElasticsearchUtils;
+  let root: Root;
+
+  beforeAll(async () => {
+    await removeLogFile();
+  });
+
+  afterAll(async () => {
+    if (root) {
+      await root.shutdown();
+    }
+    if (esServer) {
+      await esServer.stop();
+    }
+
+    await new Promise((resolve) => setTimeout(resolve, 10000));
+  });
+
+  it('migrates the documents to the highest version', async () => {
+    const { startES } = kbnTestServer.createTestServers({
+      adjustTimeout: (t: number) => jest.setTimeout(t),
+      settings: {
+        es: {
+          license: 'basic',
+          // example of original 'bar' SO:
+          // {
+          //  type: 'bar',
+          //  bar: {
+          // name: 'Mary', // may be missing
+          // surname: 'Doe, // may be missing
+          // age: 25, // may be missing
+          // old: false // may be missing
+          // },
+          //  migrationVersion: {
+          //    bar: '7.13.0',
+          //  },
+          // },
+          // SO that will fail transformation:
+          // {
+          //  type: 'space',
+          //  space: {},
+          // },
+          //
+          //
+          dataArchive: Path.join(
+            __dirname,
+            'archives',
+            '7.13.0_with_corrupt_and_transform_failures_docs.zip'
```

Comment:
FYI: dataArchive must be created with basic license https://github.com/elastic/kibana/issues/98351#issuecomment-844286396